relations/src/proof.rs

```diff
@@ -0,0 +1,102 @@
+use halo2_base::{
+    gates::{GateChip, GateInstructions},
+    poseidon::hasher::PoseidonHasher,
+    utils::BigPrimeField,
+    AssignedValue, Context,
+};
+
+use crate::poseidon_consts::{RATE, T};
+
+#[derive(Clone, Debug)]
+pub struct MerkleProof<F: BigPrimeField, const MAX_PATH_LEN: usize> {
```

Comment:
I think `MAX_PATH_LEN` is confusing, because it suggests there are paths of different lengths, while there are not. I would call it `TREE_HEIGHT`

Reply:
this is how it was done previously to allow for actual different (but bounded) trees

Reply:
like for tests or smth

Reply:
But whyyyyy :P 
I mean isn't this use-case (testing) exactly taken care of by the const generic?

Reply:
single setup for multiple circuits? but I agree it might be easier and cleaner to have a fixed size

Reply:
Done